tests/test_generate.py

```diff
@@ -15,20 +31,190 @@ def test_generator_load_screen(generator):
     assert generator.screen_components == [entity]
 
 
-def test_generator_get_screen_dimensions(generator):
+def test_generator_get_screen_dimensions_good(generator):
     test_embedded_screen = "tests/test_files/motor_embed.bob"
-
     x, y = generator._get_screen_dimensions(test_embedded_screen)
-
     assert x == 120
     assert y == 205
 
 
-# def test_build_groups(gb: Builder):
-#     generator = Generator(
-#         gb.entities, gb._gui_map, gb.components[4].name
-#     )  # TODO: remove hardcoded index
-#     generator.build_groups()
-#     with open("./tests/test_files/group.xml") as f:
-#         control = f.read()
-#     assert str(generator.group) == control
+def test_generator_get_screen_dimensions_default(generator):
+    test_embedded_screen = "tests/test_files/motor_bad.bob"
+    x, y = generator._get_screen_dimensions(test_embedded_screen)
+    assert x == 100
+    assert y == 100
+
+
+def test_generator_get_widget_dimensions_good(generator):
+    widget = Widget.EmbeddedDisplay(name="X", file="", x=0, y=0, width=205, height=120)
+
+    height, width = generator._get_widget_dimensions(widget)
+    assert height == 120
+    assert width == 205
+
+
+def test_generator_get_widget_dimensions_default(generator):
+    widget_bad = Path("tests/test_files/widget_bad.xml")
+
+    with open(widget_bad) as f:
+        xml_content_bad = f.read()
+
+    height, width = generator._get_widget_dimensions(xml_content_bad)
+    assert height == 100
+    assert width == 100
+
+
+def test_generator_get_widget_dimensions_default_2(generator):
```

Comment:
adding a `_2` on the end is not very descriptive. It would be better to just add more words, as it doesn't matter how long test function names are.
See https://github.com/bluesky/ophyd-async/blob/182780b230228d2c7bdeb97f1fc5276b94152a13/tests/unit_tests/fastcs/eiger/test_eiger_controller.py#L94 for an example of a long one

Reply:
This also applies to the test on L97 that has a `_2`